src/main/java/org/broadinstitute/hellbender/engine/ReadsDataSource.java

```diff
@@ -443,12 +443,32 @@ public SAMFileHeader getHeader() {
      */
     private SamFileHeaderMerger createHeaderMerger() {
         List<SAMFileHeader> headers = new ArrayList<>(readers.size());
+        final Set<SAMFileHeader.SortOrder> sortOrders = new HashSet<>(SAMFileHeader.SortOrder.values().length);
         for ( Map.Entry<SamReader, CloseableIterator<SAMRecord>> readerEntry : readers.entrySet() ) {
-            headers.add(readerEntry.getKey().getFileHeader());
+            final SAMFileHeader h = readerEntry.getKey().getFileHeader();
+            headers.add(h);
+            sortOrders.add(h.getSortOrder());
         }
 
-        // TODO: don't require coordinate ordering
-        SamFileHeaderMerger headerMerger = new SamFileHeaderMerger(SAMFileHeader.SortOrder.coordinate, headers, true);
+        // unknown order is removed and assumes that it is the same as the others
+        final boolean containUnknown = sortOrders.remove(SAMFileHeader.SortOrder.unknown);
```

Comment:
To clarify since it's very unclear and not really documented `unknown` is the order we get when the sort order doesn't match any of the known spec compliant orders, `unsorted` is what you get when the sort order is unspecified or is specifically marked as unsorted.  

I don't think it's a good idea to ignore unknowns in this case, they should be left in for this check.